File: CSharp/CSharp/CSharp/domain/Proba.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharp.domain
{
    public class Proba:Entity<long>
    {
        public  long Id { get; set; }
        public String Denumire { get; set; }

        public int VarstaMin { get; set; }
        public int VarstaMax { get; set; }

        public Proba(string denumire,int varstaMin, int varstaMax)
        {
            Denumire = denumire;
       
            VarstaMin = varstaMin;
            VarstaMax = varstaMax;
        }
        public Proba()
        { }
        public override bool Equals(object obj)
        {
            return obj is Proba proba &&
                   Denumire == proba.Denumire &&
                
                   VarstaMin == proba.VarstaMin &&
                   VarstaMax == proba.VarstaMax;
        }

        public override int GetHashCode()
        {
            int hashCode = 1248040347;
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(Denumire);

            hashCode = hashCode * -1521134295 + VarstaMin.GetHashCode();
            hashCode = hashCode * -1521134295 + VarstaMax.GetHashCode();
            return hashCode;
        }

    
        public override string ToString()
        {
            return string.Format("[Proba: Id={0}, Denumire={1},VarstaMin={2},VarstaMax={3}]", Id, Denumire,VarstaMin,VarstaMax);
        }
    
    }
}

File: JAVApb5/ConcursModel/src/main/java/concurs/domain/Proba.java
package concurs.domain;

import java.io.Serializable;
import java.util.Objects;

public class Proba  implements Entity<Long>, Serializable {
    private long id;
    private String denumire;
    private int varstaMin;
    private int varstaMax;

    public Proba() {

    }

    public Proba(String denumire,  int varstaMin, int varstaMax) {
        this.denumire = denumire;

        this.varstaMin = varstaMin;
        this.varstaMax = varstaMax;
    }

    public String getDenumire() {
        return denumire;
    }

    public void setDenumire(String denumire) {
        this.denumire = denumire;
    }




    public int getVarstaMin() {
        return varstaMin;
    }

    public void setVarstaMin(int varstaMin) {
        this.varstaMin = varstaMin;
    }

    public int getVarstaMax() {
        return varstaMax;
    }

    public void setVarstaMax(int varstaMax) {
        this.varstaMax = varstaMax;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Proba proba = (Proba) o;
        return id == proba.id &&
                varstaMin == proba.varstaMin &&
                varstaMax == proba.varstaMax &&
                Objects.equals(denumire, proba.denumire);
    }

    @Override
    public int hashCode() {
        return Objects.hash(denumire,varstaMin, varstaMax);
    }

    @Override
    public String toString() {
        return "Proba{" +
                "denumire='" + denumire + '\'' +

                ", varstaMin=" + varstaMin +
                ", varstaMax=" + varstaMax +
                '}';
    }
    @Override
    public Long getId() {
        return id;
    }

    @Override
    public void setId(Long id) {
        this.id=id;
    }
}
